### PR TITLE
chore: bail when no new version will be published,

### PR DESCRIPTION
**Pull request by therightstuff**
make ampqlib consume test more reliable

### CODE REVIEW

File: scripts/deploy.sh [hunk 1]
```diff
@@ -18,6 +18,19 @@ npm run build
 cp package.json dist
 setup_git
 
+echo "Dry-run of semantic-version to determine whether the package will be published"
+# merge stderr and stdout and check for "no new version is released". If found, bail.
+# don't forget to capture the grep failure with `|| echo ""` so that the script doesn't fail
+# when the line isn't found
+npm run semantic-release -- -d 2>&1 | grep "no new version is released" > no_new_version.txt || echo "It looks like a new version will be released!"
```
Comment (saartochner) [inline scripts/deploy.sh:25]: you don't need to dry run - I think that you can just validate the output of the existing command

File: scripts/deploy.sh [hunk 2]
```diff
@@ -18,6 +18,19 @@ npm run build
...
+if [ -s no_new_version.txt ]; then
+    # run the dry run a second time so that the output is printed to the console
+    echo npm run semantic-release -- -d
```
Comment (saartochner) [inline scripts/deploy.sh:28]: this is just wastefull IMO, lets simply:
`npm run semantic-release -- -d &> sematic_out.txt`
and then grep on sematic_out.txt and cat its value

**Review by saartochner — APPROVED** [review]
non mandatory comments, approved in advanced

File: scripts/deploy.sh (hunk 2 above)
Comment (therightstuff) [reply scripts/deploy.sh:28]: that's fair. it won't be as pretty, but it'll be fine like that

File: scripts/deploy.sh (hunk 1 above)
Comment (therightstuff) [reply scripts/deploy.sh:25]: that's fair 👌

File: scripts/deploy.sh (hunk 2 above)
Comment (therightstuff) [reply scripts/deploy.sh]: actually, i like your solution better, i used `tee` and it's much cleaner 🙏

**Pull request merged**